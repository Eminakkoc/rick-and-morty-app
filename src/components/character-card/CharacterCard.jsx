import React from 'react';
import PropTypes from 'prop-types';

import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import CardActionArea from '@material-ui/core/CardActionArea';

import PlaceholderImage from 'images/placeholder.jpg';

const CharacterCard = (props) => {
  const { name } = props;

  return (
    <Card className="card">
      <CardActionArea>
        <div className="card-wrapper">
          <CardContent>
            {name}
          </CardContent>
          <CardMedia
            className="card-image"
            image={PlaceholderImage}
            title="Contemplative Reptile"
          />
        </div>
      </CardActionArea>
    </Card>
  );
};

CharacterCard.propTypes = {
  name: PropTypes.string,
};

CharacterCard.defaultProps = {
  name: '',
};

export default CharacterCard;
