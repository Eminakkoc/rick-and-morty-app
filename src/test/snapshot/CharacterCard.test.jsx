// CharacterCard.test.js
import React from 'react';
import CharacterCard from '../../components/character-card/CharacterCard';
import renderer from 'react-test-renderer';

test('Character Card Component Structure', () => {
  const component = renderer.create(
    <CharacterCard
      image="image_source"
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
