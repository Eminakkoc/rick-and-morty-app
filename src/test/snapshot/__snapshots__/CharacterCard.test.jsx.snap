// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Card Component Structure 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root card MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="card-wrapper"
    >
      <div
        className="MuiCardContent-root card-info"
      >
        
      </div>
      <div
        className="MuiCardMedia-root card-image"
        title="Rick And Morty Character"
      />
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;

exports[`Link changes the class when hovered 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root card MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="card-wrapper"
    >
      <div
        className="MuiCardContent-root card-info"
      >
        
      </div>
      <div
        className="MuiCardMedia-root card-image"
        title="Rick And Morty Character"
      />
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
